lambdas/select-cri/src/main/java/uk/gov/di/ipv/core/selectcri/SelectCriHandler.java

```diff
@@ -125,21 +132,55 @@ public APIGatewayProxyResponseEvent handleRequest(
     }
 
     private APIGatewayProxyResponseEvent getNextWebJourneyCri(
-            List<VisitedCredentialIssuerDetailsDto> visitedCredentialIssuers) {
-        if (userHasNotVisited(visitedCredentialIssuers, passportCriId)) {
-            return getJourneyResponse(passportCriId);
+            List<VisitedCredentialIssuerDetailsDto> visitedCredentialIssuers, String userId)
+            throws ParseException {
+        Optional<APIGatewayProxyResponseEvent> passportResponse =
+                getCriResponse(
+                        visitedCredentialIssuers,
+                        PassportEvidenceValidator::validate,
+                        Gpg45Profile.M1A,
+                        passportCriId,
+                        userId);
+        if (passportResponse.isPresent()) {
+            return passportResponse.get();
         }
 
         if (userHasNotVisited(visitedCredentialIssuers, addressCriId)) {
             return getJourneyResponse(addressCriId);
+        } else {
+            Optional<VisitedCredentialIssuerDetailsDto> addressVisitDetails =
+                    visitedCredentialIssuers.stream()
+                            .filter(cri -> cri.getCriId().equals(addressCriId))
+                            .findFirst();
```

Comment:
```suggestion
            VisitedCredentialIssuerDetailsDto addressVisitDetails =
                    visitedCredentialIssuers.stream()
                            .filter(cri -> cri.getCriId().equals(addressCriId))
                            .findFirst().orElseThrow();
```
I think this should be safe due to the check above.